feat(aio): upgrade to CLI beta.32 and MD beta.2

wardbell opened the pull request:
**MD bug requires package.json to install from nightly instead of intended beta.2**

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

wardbell committed:
build(aio): upgrade to CLI beta.32 and MD beta.2

wardbell committed:
test(aio): redefine pipe test as a pending test to confirm code example

gkalpak commented on aio/e2e/app.po.ts:
```diff
@@ -3,7 +3,7 @@ import { browser, element, by } from 'protractor';
 
   links = element.all(by.css('md-toolbar a'));
-  datePipeLink = element(by.css('md-toolbar a[aioNavLink="api/common/date-pipe"]'));
+  datePipeLink = element.all(by.css('md-toolbar a[aioNavLink="api/common/date-pipe"]'));
```
Why are there multiple nav-links for the same URL?

wardbell replied:
It appears twice in the DOM, once in the top menu bar and once at the top of the left nav. Only one set of links is visible to the user at any one time. It's a hide/show thing.

gkalpak replied:
Not feeling strong about it, but I would prefer to have the `.get(0)` here, rather than on call site. The fact that there are multiple navs for the same target is an implementation detail that doesn't need to "leak" outside `app.po.ts` imo.

gkalpak commented on aio/package.json:
```diff
@@ -9,55 +9,59 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "build": "ng build",
```
What happened to `build`?

wardbell replied:
Restoring

gkalpak commented on aio/package.json:
```diff
@@ -9,55 +9,59 @@
+    "e2e": "ng e2e",
+
+    "lint:old": "tslint \"src/**/*.ts\" --project src/tsconfig.json --type-check && tslint \"e2e/**/*.ts\" --project e2e/tsconfig.json --type-check",
     "pree2e": "webdriver-manager update --standalone false --gecko false",
```
Should this be renamed to `pree2e:old`? Is it still needed before `ng e2e`?

wardbell replied:
For reasons unknown to me, the `e2e` command does not update webdriver even thought the command name is `pree2e`.  The latter command must be run by hand after every `npm install` or `yarn install`.

gkalpak replied:
That's is weird (and most probably untrue 😛).

gkalpak replied:
For some reason, both ChromeDriver v2.15 and v2.27 are downloaded on Travis (e.g. https://travis-ci.org/angular/angular/jobs/203945848). No idea why or if this is related...

gkalpak replied:
Nit: It is clearer imo to keep `pre-`/`post-` scripts close to the main script (so that would be moving this right above `e2e`).

gkalpak commented on aio/protractor.conf.js:
```diff
@@ -2,19 +2,15 @@
   capabilities: {
-    browserName: 'chrome',
-    chromeOptions: {
-      binary: process.env.CHROME_BIN
```
Why is this no longer necessary?

wardbell replied:
Restoring

petebacondarwin reviewed (changes requested):
Can you squash in this commit: https://github.com/IdeaBlade/angular/pull/1/commits/320a2df752a6f310261daa7b1551134249b1c7bf

petebacondarwin reviewed (approved):
LGtM

IgorMinar commented on scripts/ci-lite/env.sh:
```diff
@@ -7,7 +7,7 @@ echo 'travis_fold:start:ENV'
 NPM_VERSION=3.10.7 # do not upgrade to >3.10.8 unless https://github.com/npm/npm/issues/14042 is resolved
 YARN_VERSION=0.19.1
-CHROMIUM_VERSION=403382 # Chrome 53 linux stable, see https://www.chromium.org/developers/calendar
+CHROMIUM_VERSION=433059 # Chrome 56 linux stable, see https://www.chromium.org/developers/calendar
```
why this change? this is a big change for our CI that must be in a separate commit.

wardbell replied:
Moved to separate commit. From now on, any changes to anything OUTSIDE of the `aio` will be in a separate commit and possibly a separate PR.

IgorMinar commented on scripts/ci-lite/test_aio.sh:
```diff
@@ -50,7 +50,7 @@ echo 'travis_fold:end:test.aio.unit'
 # Run e2e tests
 cd ../../aio
-yarn start &
+# e2e starts server itself as of beta.32.3
```
no need for this comment. it's not going to mean anything to us in 6 months.

wardbell replied:
Will remove

wardbell replied:
Should this script file, which is aio-specific, be in the aio folder?

IgorMinar commented on aio/karma.conf.js:
```diff
@@ -7,9 +7,13 @@ module.exports = function (config) {
       require('karma-jasmine'),
       require('karma-chrome-launcher'),
-      require('karma-remap-istanbul'),
+      require('karma-jasmine-html-reporter'),
```
this change is not mentioned in the commit message. it should be a separate commit, but you already have a PR for this. was this left here by accident?

wardbell replied:
It will obviate the need for that PR. This change is now part of CLI beta.32.3

wardbell replied:
I'll close that PR #14392 when this PR merges.

IgorMinar commented on aio/package.json:
```diff
@@ -10,54 +10,57 @@
-    "e2e": "protractor",
+    "e2e": "ng e2e --no-webdriver-update",
+    "lint:old": "tslint \"src/**/*.ts\" --project src/tsconfig.json --type-check && tslint \"e2e/**/*.ts\" --project e2e/tsconfig.json --type-check",
+    "e2e:old": "protractor",
```
do we still need the old lint and e2e tasks?

wardbell replied:
I was being cautious. I don't think so. Will remove.

IgorMinar commented on aio/tslint.json:
```diff
@@ -46,7 +46,7 @@
     "no-empty-interface": true,
     "no-eval": true,
-    "no-inferrable-types": true,
+    "no-inferrable-types": [true, "ignore-params"],
```
why this change?

wardbell replied:
It's in cli: https://github.com/angular/angular-cli/blob/master/packages/%40angular/cli/blueprints/ng/files/tslint.json#L49

IgorMinar replied:
thanks for explanation

wardbell committed:
ci: update local chrome to v56

Change required in previous commit ab96257bc that upgrades the CLI version.
env.sh: protractor v.5 requires a later version of chromium. Upping to 433059/Chrome 56.

IgorMinar commented on scripts/ci-lite/env.sh:
```diff
@@ -7,7 +7,7 @@ echo 'travis_fold:start:ENV'
 NODE_VERSION=6.9.5
```
the commit message should no longer be scoped to "aio" because this commit affects stuff outside of the `aio/` directory. Suggested replacement: `ci: update local chrome to v56` + merge the `test_aio.sh` change in to the previous commit that upgrades the CLI version.

wardbell replied:
Changing that now
